### PR TITLE
feature: adding ExclusiveMinimum and ExclusiveMaximum

### PR DESCRIPTION
**Pull request by selmison**
PR of issue #2726

### CODE REVIEW

File: expr/example.go [hunk 1]
```diff
@@ -236,9 +239,15 @@ func byMinMax(a *AttributeExpr, r *Random) interface{} {
...
+	}
 	if a.Validation.Maximum != nil {
 		max = *a.Validation.Maximum
 	}
+	if a.Validation.ExclusiveMinimum != nil {
+		min = *a.Validation.ExclusiveMinimum
+	}
 	if a.Validation.Minimum != nil {
 		min = *a.Validation.Minimum
 	} else {
```
Comment (raphael) [inline expr/example.go:251]: We need to test that there is no exclusive minimum as well here, something like:
```go
} else if a.Validation.ExclusiveMinimum == nil {
```

**Review by raphael — CHANGES_REQUESTED** [review]
Thank you, this is great! Just one issue reported below.

**Review by raphael — APPROVED** [review]

**Comment by raphael** [comment]
This is great, thank you!

**Comment by raphael** [comment]
One last request: could you back-port this to the `v2` branch please? Simply cherry-pick the merge commit. Thanks again!
